src/compas/datastructures/mesh/_mesh.py

```diff
@@ -824,6 +824,40 @@ def to_obj(self, filepath):
                 vertices = [key_index[key] + 1 for key in vertices]
                 fh.write(' '.join(['f'] + [str(index) for index in vertices]) + '\n')
 
+    def to_obj_with_attribute(self, filepath, attribute):
```

Comment:
name doesn't really says what it does in my opinion.
also, should this not be part of the `OBJ` API?
then there could just be a flag in the `to_obj` function of the mesh to request "grouping by attribute"...

Reply:
i considered adding this functionality to the existing `to_obj` function, but i thought this should maybe stay simple and lightweight. with OBJ API, you mean `compas.files.obj.OBJ`?

Reply:
i more meant that the entire implementation should go to `compas.files.OBJ`, which is actually also true for the current `to_obj` functionality. the mesh just needs to provide the `OBJ` with data it understands after which it an do its thing with it...

now the mesh also takes care of the actual writing of the OBJ, which should not be the case in either of the two functions...

Reply:
ok I see. I didn't know `to_obj` was supposed to be moved anyway and therefore built on top of it. but having an `OBJ` object take care of file writing makes sense, optionally with a flag for face grouping! Currently, there are the classes `OBJ`, `OBJReader` and `OBJParser`. Would there be a new class `OBJWriter` then?

Reply:
This could of course also be setup for all the different `to_*` methods, preferably using a common superclass named e.g. BaseWriter..

Reply:
agreed. it was always the goal to have writers as well. i just never got around to it...

Reply:
I'll add that to my list of things to do. (After current PR and refactor of binary read)